src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -738,7 +744,12 @@ public void valueChanged(ListSelectionEvent evt) {
                         setDirectorySelected(true);
                         setDirectory(file);
                         if (usesSingleFilePane) {
-                            chooser.setSelectedFile(null);
+                            Path path = Paths.get(file.getPath());
+                            if (Files.isSymbolicLink(path)) {
```

Comment:
```suggestion
                            if (Files.isSymbolicLink(file.toPath())) {
```

One more instance of `Paths.get` should be replaced with `file.toPath()`.